### DISCUSSION TITLE
Question about design limits to ord_calc and ord_plot

### DISCUSSION THREAD

**Discussion by davised** [david-barnett/microViz#103] [General]
Hi David,

Working on a new microbiome analysis so I'm back again with questions.

1. Why is the formula interface not provided when using ord_calc constrained analyses? I suppose it's so you can automatically plot the constraints and conditions. How might we implement that?
2. As a follow up, why do you not allow factors as constraints/conditions? This is allowed in the vegan functions, and passing them through the phyloseq `ordinate()` function indeed works.
3. I noticed you do not allow/provide plotting taxa when doing, e.g. PCoA. This is possible using the underlying phyloseq, e.g.
```R
ps %>% 
  tax_transform("compositional") %>%
  otu_get() %>%
  dist("manhattan") %>%
  ordinate(ps, "PCoA", .) %>%
  plot_ordination(ps, ., type = "biplot")
```

You could even advise folks, by including a `sqrt.dist` as in `vegan::capscale` to convert dissimilarities such that they do not produce negative eigenvalues. Or, suggest compositional transform for Bray-Curtis to equal manhattan distance, as I do above.

Looking forward to this discussion.

Cheers,

Ed

**Comment by david-barnett** [comment]
hi Ed, nice to hear from you again

Some thoughts for now

1. It should be quite possible to provide a formula interface (or rather, passing a string containing a formula) using something like all.vars or some regex processing (the difficult bit would be probably be getting the special `+ condition(vars)` syntax right). I just never needed it or get a request for it before so I didn't make it. Why do you want it?

2. Correctly handling factors seemed more tricky at the time I made ord_calc and ord_plot, particularly ord_plot with its automatic vector labelling. My first instinct would be to implement something that just converts factors into dummy variables with reference coding (as integers). The result of that would be usable directly with the current code. Currently the user can just code the dummy variables themselves. (One problem with that being firstly that then the user needs to know you need to do this, and secondly that you shouldn't do this when running permanova to find that factor's overall significance). Another reason I didn't handle this is that vegan plotting seems to prefer plotting some kind of centroid position symbols for factor levels rather than plotting arrows, and I haven't thought carefully about why, and whether microViz should also do this. 

3. It may not be a good idea with anything other than Euclidean distances, so I am unsure if I would want to implement this. See here https://github.com/vegandevs/vegan/issues/254
I think that adding something that allows you to manually add correlation-like arrows for supplementary variables (taxa or whatever environmental variables) after doing the ordination (e.g. using envfit) is probably safer, in the sense of not risking encouraging misuse/misinterpretation
https://www.davidzeleny.net/anadat-r/doku.php/en:suppl_vars_examples 

As a more general note - I'm not going to be adding any new features before May, as I need to focus on my PhD thesis.
The good news is that I am probably going to have a job afterwards where I am again able to focus some substantial portion of my time on software development including microViz.

cheers
David
